flocker/node/test/test_script.py

```diff
@@ -229,6 +234,58 @@ def test_initialized(self):
             ),
         )
 
+    def test_logging(self):
+        """
+        Logging is setup by a logging stanza.
+        """
+        # Save and restore root logger state
+        def restore_logger(logger, level, handlers):
+            logger.setLevel(level)
+            for handler in logger.handlers[:]:
+                logger.removeHandler(handler)
+            for handler in handlers:
+                logger.addHandler(handler)
+
+        logger = logging.getLogger()
+        self.addCleanup(
+            restore_logger, logger, logger.getEffectiveLevel(),
+            logger.handlers[:]
+        )
+
+        # Setup an AgentService with a logging stanza
+        host = b"192.0.2.13"
+        port = 2314
+        name = u"from_config-test"
+        logfile = self.make_temporary_directory().child('logfile')
```

Comment:
Yeah, I tried that initially. But because this test then goes on to call `setup_config` which uses `mktemp()`, it seemed like I was creating a file and a directory with the same name. Since it actually seemed to work, maybe I was mistaken, but I went with a safer version.